src/containers/SignInPage/SignInPage.tsx

```diff
@@ -0,0 +1,103 @@
+import * as React from 'react'
+
+import { SignIn, SignInI18N } from 'decentraland-ui'
+
+import { SignInPageProps, SignInState } from './SignInPage.types'
+import { T } from '../../modules/translation/utils'
+import { isMobile } from '../../lib/utils'
+
+export default class SignInPage extends React.PureComponent<
+  SignInPageProps,
+  SignInState
+> {
+  constructor(props: SignInPageProps) {
+    super(props)
+    this.state = {
+      hasError: false
+    }
+  }
+
+  componentWillReceiveProps(nextProps: SignInPageProps) {
+    if (nextProps.hasError && !this.state.hasError) {
+      this.setState({
+        hasError: true
+      })
+    } else if (!nextProps.hasError && this.state.hasError) {
+      this.setState({
+        hasError: false
+      })
```

Comment:
mm not quite the same, because if `nextProps.hasError === true` and `this.state.hasError === true` we want the state to remain `true` (ie, if any of the other props change).

Reply:
Gotcha! I wasn't sure about that state 👍